### PR TITLE
SS-1206 Site dir option should have path validation

### PR DESCRIPTION
**Pull request by anondo1969**
## Description

Source: https://scilifelab.atlassian.net/browse/SS-1206

The site-dir option in shinyproxy apps should disallow  paths/like\ this . This would fail in the helm deployment.

We’d like side-dir to be compatible with k8s labels specification [Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) 

Valid label value:

1. must be 63 characters or less (can be empty),
 
2. unless empty, must begin and end with an alphanumeric character ([a-z0-9A-Z]),

3. could contain dashes (-), underscores (_), dots (.), and alphanumerics between.

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have included migration files (if there are changes to the model classes)
- [x] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

## Further comments

Anything else you think we should know before merging your code!

### CODE REVIEW

File: apps/helpers.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 from enum import Enum
 from typing import Optional
 
-from django.core.exceptions import ObjectDoesNotExist
+import regex as re
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
```
Comment (churnikov) [inline apps/helpers.py:6]: Isort was not applied here

File: apps/helpers.py [hunk 2]
```diff
@@ -371,3 +372,25 @@ def save_instance_and_related_data(instance, form):
     instance.set_k8s_values()
     instance.url = get_URI(instance)
     instance.save(update_fields=["k8s_values", "url"])
+
+
+def validate_path_k8s_label_compatible(candidate: str):
```
Comment (churnikov) [inline apps/helpers.py:377]:
```suggestion
def validate_path_k8s_label_compatible(candidate: str) -> None:
```

**Review by churnikov — APPROVED** [review]
After my two comments lgtm to merge

File: apps/helpers.py (hunk 2 above)
Comment (anondo1969) [reply apps/helpers.py]: Thanks. Applied.

File: apps/helpers.py (hunk 1 above)
Comment (churnikov) [reply apps/helpers.py:6]: https://scilifelab.slack.com/archives/C02L5Q7HWR1/p1732872632171369?thread_ts=1732623632.294409&cid=C02L5Q7HWR1

**Review by churnikov — APPROVED** [review]

**Pull request merged**